### PR TITLE
Converted all sprintf to snprintf and fixed many warnings too

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Review by hjmjohnson — APPROVED** [review]
@seanm Thank you for your efforts.  These all look good to me.  It's nice to plug these code linting warnings.

**Comment by afni-rickr** [comment]
@seanm @hjmjohnson This looks great.
The only question in my mind is for nifti1_magic and nifti2_magic.  This change removes those from the library.  Since they were not explicitly exported in the header file, that might be justified.  But still, it removes symbols from the library.
I don't have a strong preference, but perhaps putting them in nifti2_io.h would be safer, in case someone is using one or both.
Preferences?  Thanks!

**Comment by seanm** [comment]
>...in case someone is using one...

How could anyone be using it (as it was not in the header)?

**Comment by afni-rickr** [comment]
The symbol is in the library, so it can be used.  There just isn't any declaration provided (by us).  To avoid a warning for such use (or maybe to get proper use), one would add their own declaration.

**Comment by afni-rickr** [comment]
After a little pondering, I am leaning towards keeping them in the library and adding externs to the header.  Since they are an explicit part of the NIFTI standard, it makes sense for the library to provide defined symbols (though we had no such symbols in the NIFTI-1 version).  And since they were already there, that adds weight to not removing them.
So @seanm, would you be okay keeping char * nifti[12]_magic in nifti2_io.c and then adding extern declarations in nifti2_io.h?  Since you have done all of this, it might be nice for you to do that, too.  Otherwise, I would be happy to.  It is up to you.
Thanks again!

**Comment by seanm** [comment]
@afni-rickr sounds good; done.

**Comment by seanm** [comment]
There are a few other warnings I'd like to fix, but didn't do in this PR since they require some discussion... but now that we are discussing... there are several like:

```
/nifti_clib/niftilib/nifti1.h:17:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _NIFTI_HEADER_
        ^
```

This pedantic warning is due to underscore + uppercase being reserved.  We can of course just change the spelling, but do you consider this symbol public API in some sense?  Think other code is checking this #define?

**Review by afni-rickr — APPROVED** [review]

**Comment by afni-rickr** [comment]
I did not actually realize that, thanks.  It seems those warnings were just recently added.  The macros are only defined to prevent nested inclusion of the header files.  The naming is a style we adopted that applies to hundreds of files.  <sigh>  :)
Anyway, it does seem like a good idea to change that.  Maybe putting 'h' in the front would be a good rule.
But let's ponder that in a different pull request (if you care to do it).  This one seems ready to merge.
Does that sound good?  Thanks again.

File: nifti2/nifti2_io.c [hunk 1]
```diff
@@ -418,8 +418,7 @@ static nifti_global_options g_opts = {
         0, /* alter_cifti       - alter CIFTI dims to use nx,t,u,v*/
 };
 
-char nifti1_magic[4] = { 'n', '+', '1', '\0' };
-char nifti2_magic[8] = { 'n', '+', '2', '\0', '\r', '\n', '\032', '\n' };
+static char nifti2_magic[8] = { 'n', '+', '2', '\0', '\r', '\n', '\032', '\n' };
```
[the remark this answers is not in the document]
Comment (seanm) [reply nifti2/nifti2_io.c]: Yeah, that does look suspicious. But a behaviour change should be in a different commit I think, and in any case I don't know this code at all really, so I leave that to you :)

**Comment by seanm** [comment]
> This one seems ready to merge.

Great!   I'll do the _UPPER fixes in a new PR.

**Comment by afni-rickr** [comment]
Great, thanks!

File: nifti2/nifti2_io.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (afni-rickr) [reply nifti2/nifti2_io.c]: Particularly with the '\0' in the middle, use of these will probably stick to being done via memcpy() and memcmp().  So maybe that is why the termination was not considered so important.

**Pull request merged**